src/views/Drawer/DrawerSidebar.js

```diff
@@ -59,26 +68,51 @@ class DrawerSidebar extends PureComponent<void, Props, void> {
     return route.routeName;
   };
 
-  _renderIcon = ({ focused, tintColor, route }: DrawerScene) => {
-    const drawer = this._getScreenConfig(route.key, 'drawer');
-    if (drawer && drawer.icon) {
-      return drawer.icon({
-        tintColor,
-        focused,
-      });
-    }
-    return null;
+  _renderIcon = (configs: DrawerConfigs) => ({ focused, route }: DrawerScene) => {
+    const drawer = focused ? configs[route.key].focused : configs[route.key].unfocused;
+    return drawer && drawer.icon ? drawer.icon : null;
   };
 
   render() {
-    const ContentComponent = this.props.contentComponent;
+    const {
+      contentComponent: ContentComponent,
+      contentOptions,
+      childNavigationProps,
+      router,
+      navigation,
+    } = this.props;
+
+    const defaultDrawerProps = ContentComponent.defaultProps || {};
+    const activeTintColor =
+      contentOptions && contentOptions.activeTintColor ?
+      contentOptions.activeTintColor :
+      defaultDrawerProps.activeTintColor;
+    const inactiveTintColor =
+      contentOptions && contentOptions.inactiveTintColor ?
+      contentOptions.inactiveTintColor :
+      defaultDrawerProps.inactiveTintColor;
+
+    const configs = navigation.state.routes.reduce((acc: DrawerConfigs, route: *) => {
```

Comment:
If you pass the router down then you could handle this stuff in the content component. This seems appropriate because many content components may be custom and wont need all this stuff